### PR TITLE
feat(md): handle BLOCKED state for constraints

### PR DESCRIPTION
**Pull request by ranihorev**
This mainly affects stateful constraints before they can be acted upon by the user, and therefore we want to hide the actions
image

### CODE REVIEW

File: app/scripts/modules/core/src/managed/overview/artifact/VersionOperation.tsx [hunk 1]
```diff
@@ -3,26 +3,37 @@ import React from 'react';
...
-  PENDING: { icon: 'far fa-hourglass', displayName: 'pending' },
-  NOT_EVALUATED: { icon: 'far fa-hourglass', displayName: 'pending' },
+  PENDING: { icon: DEFAULT_ICON, displayName: 'pending' },
+  NOT_EVALUATED: { icon: DEFAULT_ICON, displayName: 'pending' },
+  BLOCKED: { icon: DEFAULT_ICON, displayName: 'pending' },
 };
 
+// This forces the consumer to verify that the value exists (important in the case of a breaking change on the backend)
+export const actionStatusUtils: Partial<ActionStatusUtils> = actionStatusUtilsInternal;
+
```
Comment (caseyhebebrand) [inline app/scripts/modules/core/src/managed/overview/artifact/VersionOperation.tsx:27]: Can you explain this a bit more? I don't understand what the reassignment does here.

Comment (ranihorev) [reply app/scripts/modules/core/src/managed/overview/artifact/VersionOperation.tsx:27]: You can see below where I use the object the return type of this `actionStatusUtils[status]` can be undefined and I have to use `?`. By adding this I'm making sure that adding new options to the backend won't cause the frontend to crash

Comment (caseyhebebrand) [reply app/scripts/modules/core/src/managed/overview/artifact/VersionOperation.tsx:27]: I don't understand why you can't use `actionStatusUtilsInternal[status]?.icon` If a new status is added to the backend, then `actionStatusUtilsInternal[status]` would also be undefined, and the optional chaining would serve the same purpose.

Comment (ranihorev) [reply app/scripts/modules/core/src/managed/overview/artifact/VersionOperation.tsx:27]: That's exactly what I'm doing, the purpose of line 26 is to force you to do that :)

Comment (caseyhebebrand) [reply app/scripts/modules/core/src/managed/overview/artifact/VersionOperation.tsx:27]: I am not seeing a difference between `actionStatusUtilsInternal[status]?.icon` and `actionStatusUtils[status]?.icon`, so I don't think that line 26 is necessary.

Comment (ranihorev) [reply app/scripts/modules/core/src/managed/overview/artifact/VersionOperation.tsx:27]: The difference is:
- `actionStatusUtilsInternal[status].icon` is valid and won't raise an error
- `actionStatusUtils[status].icon` this is not valid and therefore forces the consume to handle the undefined case properly

Comment (caseyhebebrand) [reply app/scripts/modules/core/src/managed/overview/artifact/VersionOperation.tsx:27]: But aren't they the same thing? This line just says `actionStatusUtils = actionStatusUtilsInternal`

**Review by caseyhebebrand — APPROVED** [review]

**Pull request merged**